src/main/java/org/broadinstitute/hellbender/tools/walkers/genotyper/IndependentSampleGenotypesModel.java

```diff
@@ -0,0 +1,69 @@
+package org.broadinstitute.hellbender.tools.walkers.genotyper;
+
+import htsjdk.variant.variantcontext.Allele;
+import htsjdk.variant.variantcontext.GenotypeLikelihoods;
+import org.broadinstitute.hellbender.utils.Utils;
+import org.broadinstitute.hellbender.utils.genotyper.AlleleList;
+import org.broadinstitute.hellbender.utils.genotyper.AlleleListPermutation;
+import org.broadinstitute.hellbender.utils.genotyper.LikelihoodMatrix;
+
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * This class delegates genotyping to allele count- and ploidy-dependent {@link GenotypeLikelihoodCalculator}s
+ * under the assumption that sample genotypes are independent conditional on their population frequencies.
+ */
+public final class IndependentSampleGenotypesModel {
```

Comment:
This class has had quite a makeover -- could any of the changes you made possibly affect `HaplotypeCaller` results? Have you run the `HaplotypeCallerIntegrationTest` in gatk-protected using a snapshot of this branch?